azurerm/resource_arm_managed_cluster.go

```diff
@@ -0,0 +1,556 @@
+							ValidateFunc: validation.IntBetween(1, 100),
+						},
+
+						"dns_prefix": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"fqdn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"vm_size": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"os_disk_size_gb": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  0,
+						},
+
+						"storage_profile": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(containerservice.StorageAccount),
+								string(containerservice.ManagedDisks),
+							}, true),
+						},
+
+						"vnet_subnet_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"os_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  containerservice.Linux,
+						},
+					},
+				},
+				Set: resourceAzureRMManagedClusterAgentPoolProfilesHash,
+			},
+
+			"service_principal": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"client_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"client_secret": {
+							Type:      schema.TypeString,
+							Required:  true,
+							Sensitive: true,
+						},
+					},
+				},
+				Set: resourceAzureRMManagedClusterServicePrincipalProfileHash,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmManagedClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	managedClusterClient := client.managedClustersClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM AKS managed cluster creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	dnsPrefix := d.Get("dns_prefix").(string)
+	kubernetesVersion := d.Get("kubernetes_version").(string)
+
+	linuxProfile := expandAzureRmManagedClusterLinuxProfile(d)
+	agentProfiles := expandAzureRmManagedClusterAgentProfiles(d)
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	parameters := containerservice.ManagedCluster{
+		Name:     &name,
+		Location: &location,
+		ManagedClusterProperties: &containerservice.ManagedClusterProperties{
+			DNSPrefix:         &dnsPrefix,
+			KubernetesVersion: &kubernetesVersion,
+			LinuxProfile:      &linuxProfile,
+			AgentPoolProfiles: &agentProfiles,
+		},
+		Tags: expandTags(tags),
+	}
+
+	servicePrincipalProfile := expandAzureRmManagedClusterServicePrincipal(d)
+	if servicePrincipalProfile != nil {
+		parameters.ServicePrincipalProfile = servicePrincipalProfile
+	}
+
+	ctx := client.StopContext
+	_, error := managedClusterClient.CreateOrUpdate(ctx, resGroup, name, parameters)
+	if error != nil {
+		return error
+	}
```

Comment:
CreateOrUpdate returns a future - so we should be able to make this:

```
future, err :=  managedClusterClient.CreateOrUpdate(ctx, resGroup, name, parameters)
if err != nil {
  return err
}

err = future.WaitForCompletion(ctx, managedClusterClient.Client)
if err != nil {
  return err
}
```

(which means we should be able to remove the poller below)

Reply:
Thanks, fixed.